examples/nlp/dialogue/preprocess_for_zero_shot_slot_filling.py

```diff
@@ -0,0 +1,199 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import glob
+import os
+import shutil
+
+
+def read_data(args):
+    preprocess_file_path = args.preprocess_file_path
+    if args.dataset == 'conll_2003':
+        with open(preprocess_file_path + '/text_train.txt') as f:
+            train = f.read().strip().split('\n')
+        train = ["sentence\tlabel"] + train
+        with open(preprocess_file_path + '/labels_train.txt') as f:
+            train_slots = f.read().strip().split('\n')
+        with open(preprocess_file_path + '/text_test.txt') as f:
+            test = f.read().strip().split('\n')
+        test = ["sentence\tlabel"] + test
+        with open(preprocess_file_path + '/labels_test.txt') as f:
+            test_slots = f.read().strip().split('\n')
+    else:
+        with open(preprocess_file_path + '/train.tsv') as f:
+            train = f.read().strip().split('\n')
+        with open(preprocess_file_path + '/train_slots.tsv') as f:
+            train_slots = f.read().strip().split('\n')
+        with open(preprocess_file_path + '/test.tsv') as f:
+            test = f.read().strip().split('\n')
+        with open(preprocess_file_path + '/test_slots.tsv') as f:
+            test_slots = f.read().strip().split('\n')
+    return train, train_slots, test, test_slots
+
+
+def creat_folder_and_copy_files(args, output_path):
+    os.makedirs(output_path)
+    print("The new directory " + output_path + " is created!")
+
+    # all dataset need to copy "dict.slots.csv"
+    copy_file(args.preprocess_file_path, output_path, "dict.slots.csv")
+
+    # only 'assistant' and 'drive_through' dataset need to copy "dict.intents.csv"
+    if args.dataset in ['assistant', 'drive_through']:
+        copy_file(args.preprocess_file_path, output_path, "dict.intents.csv")
+
+
+def copy_file(source_path, target_path, filename):
+    global_parameters = "*"
+    filename_with_path = source_path + filename
+
+    if filename_with_path in glob.glob(os.path.join(source_path, global_parameters)):
+        if filename_with_path not in glob.glob(os.path.join(target_path, global_parameters)):
+            shutil.copy(filename_with_path, target_path)
+            print("The file " + filename_with_path + " is copied!")
+        else:
+            print("{} exists in {}".format(filename_with_path, os.path.join(os.path.split(target_path)[-2:])))
+
+
+def from_slot_name_to_slot_id(train, train_slots):
+
+    # manual build dic
+    dict_slots = []
+    dict_slots = ['O', 'B-LOC', 'I-LOC', 'B-MISC', 'I-MISC', 'B-ORG', 'I-ORG', 'B-PER', 'I-PER']
+
+    dict_slots_id = {}
```

Comment:
This next three lines can be combined into `dict_slots_id = {d: str(idx) for idx, d in enumerate(dict_slots)}`